src/components/Onboarding/kit/FileUploaderField/index.js

```diff
@@ -0,0 +1,213 @@
+      }
+    },
+    [maxFileSize, onFileUpdated]
+  )
+  const {
+    getRootProps,
+    getInputProps,
+    isDragAccept,
+    isDragReject,
+  } = useDropzone({
+    accept: allowedMIMETypes,
+    onDrop,
+    multiple: false,
+  })
+  const { mainColor, backgroundColor } = getDropzoneColor(
+    theme,
+    isDragReject,
+    isDragAccept
+  )
+
+  return (
+    <div
+      css={`
+        width: 100%;
+        color: ${theme.contentSecondary};
+        ${textStyle('body2')};
+      `}
+    >
+      <Field label={label} required={required}>
+        <div
+          css={`
+            margin-top: ${1 * GU}px;
+          `}
+        >
+          {description || (
+            <div>
+              Drag and drop a file here or <TextFileUploader />.
+            </div>
+          )}
+        </div>
+        {validExtensions && (
+          <div
+            css={`
+              margin-top: ${1 * GU}px;
+              color: ${mainColor};
+              font-weight: bold;
+            `}
+          >
+            Valid file format: {toOxford(validExtensions)}
+          </div>
+        )}
+      </Field>
+      {/* Set dropzone here due to some problems with the field component and the onclick event bubbling. */}
+      <Card
+        width="100%"
+        height="135px"
+        css={`
+          border-style: dashed;
+          background-color: ${backgroundColor};
+          border-color: ${mainColor};
+          margin-top: ${-1 * GU}px;
+          margin-bottom: ${3 * GU}px;
+          transition: background-color 0.5s;
+        `}
+      >
+        <div
+          {...getRootProps({ onClick: e => e.stopPropagation() })}
+          css={`
+            height: 100%;
+            width: 100%;
+            padding: ${1 * GU}px;
+          `}
+        >
+          <div
+            css={`
+              width: 100%;
+              height: 100%;
+              display: flex;
+              justify-content: center;
+              align-items: center;
+              transition-timing-function: ease-in;
+              border: 0px solid #d0d0d0;
+            `}
+          >
+            <input
+              {...getInputProps({
+                id: FILE_UPLOADER_ID,
+                /* Include file extensions as well due to some MIME
+                 types aren't working on input tag, e.g., text/markdown. */
+                accept: allowedMIMETypes
+                  ? [...allowedMIMETypes, ...validExtensions]
+                  : null,
+              })}
+            />
+            {!file ? (
+              <div
+                css={`
+                  border-radius: 50%;
+                  background-color: ${mainColor};
+                  width: ${iconSize}px;
+                  height: ${iconSize}px;
+                  display: flex;
+                  justify-content: center;
+                  align-items: center;
+                  transition: background-color 0.5s;
+                `}
+              >
+                <IconUpload
```

Comment:
Maybe it is difficult to implement. But my first impulse was clicking the icon to browse the files. Should we have an `onClick` handler with similar logic to the `TextFileUploader`?

Reply:
It looks like a button so I think it's ok to display the file browser dialog when clicking it